activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/ProcessVariablesHelper.java

```diff
@@ -0,0 +1,128 @@
+   
+    private final Map<String, ProcessExtensionModel> processExtensionModelMap;
+    private final VariableValidationService variableValidationService;
+    private final DateFormatterProvider dateFormatterProvider;
+ 
+    public ProcessVariablesHelper(DateFormatterProvider dateFormatterProvider,
+                                  Map<String, ProcessExtensionModel> processExtensionModelMap,
+                                  VariableValidationService variableValidationService) {
+        this.dateFormatterProvider = dateFormatterProvider;
+        this.processExtensionModelMap = processExtensionModelMap;
+        this.variableValidationService = variableValidationService;
+    }
+    
+    private Optional<Map<String, VariableDefinition>> getVariableDefinitionMap(String processDefinitionKey) {
+        ProcessExtensionModel processExtensionModel = processExtensionModelMap.get(processDefinitionKey);
+
+        return Optional.ofNullable(processExtensionModel)
+                .map(ProcessExtensionModel::getExtensions)
+                .map(Extension::getProperties);
+    }
+    
+    public List<ActivitiException> checkPayloadVariables(Map<String, Object> variablePayloadMap,
+                                                         String processDefinitionKey,
+                                                         boolean validate) {
+        
+        final String errorMessage = "Variable with name {0} does not exists.";
+        final String errorDateTimeParse = "Error parsing date/time variable with a name {0}: {1}";
+        
+        final Optional<Map<String, VariableDefinition>> variableDefinitionMap = getVariableDefinitionMap(processDefinitionKey);
+        List<ActivitiException> activitiExceptions = new ArrayList<>();
+        
+        if (variableDefinitionMap.isPresent()) {
+                    
+            for (Map.Entry<String, Object> var : variablePayloadMap.entrySet()) {
+                String name = var.getKey();
+                Object value = var.getValue();
+                boolean found = false;
+                for (Map.Entry<String, VariableDefinition> entry : variableDefinitionMap.get().entrySet()) {
+                    
+                    if (entry.getKey().equals(name)) {
+                        String type = entry.getValue().getType();
+                        found = true;
+                        
+                        if ("date".equals(type) &&  value != null) {
+                            try {
+                                Date d = dateFormatterProvider.convert2Date(value);
+                                if (d != null) {
+                                    var.setValue(value = d);
+                                }   
+                            } catch (Exception e) {
+                                activitiExceptions.add(new ActivitiException(MessageFormat.format(errorDateTimeParse, name, e.getMessage())));
+                            }
+                        } else {
```

Comment:
IMHO, If I get a type date, then I know that the type should be date, and if conversion fails - exception... Why to validate this again?
But I agree that this can be rewritten better...